### PR TITLE
branch-2.1: [fix](sql) Fix error result for column tosql #44997

### PR DESCRIPTION
**Pull request by wyxxxcat**
Cherry-picked from https://github.com/apache/doris/pull/44997

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by wyxxxcat** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 1]
```diff
@@ -600,7 +590,21 @@ public String toSql() {
         }
 
         if (defaultValue.isSet) {
-            sb.append(defaultValue.toSql()).append(" ");
+            if (defaultValue.value != null) {
+                if (typeDef.getType().getPrimitiveType() != PrimitiveType.BITMAP
```
Comment (Alex1888) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:594]: nested if is not a good pattern here. try to use 
```
if (defaultValue.value = null){
}
eles if(typeDef.getType().getPrimitiveType() == PrimitiveType.BITMAP) {
}

```

**Comment by wyxxxcat** [comment]
run feut

**Comment by wyxxxcat** [comment]
run buildall

**Pull request merged**